src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/resource/SyncAndMigrateResourceTest.kt

```diff
@@ -270,13 +269,106 @@ class SyncAndMigrateResourceTest : SqsIntegrationTestBase() {
         .isBadRequest
     }
 
+    @Test
+    fun `can sync non-association by ID`() {
+      val existingNa = repository.save(
+        genNonAssociation(
+          firstPrisonerNumber = "A7777XT",
+          secondPrisonerNumber = "B7777XT",
+        ),
+      )
+      val request = UpsertSyncRequest(
+        id = existingNa.id,
```

Comment:
@ushkarev this one uses the ID

Reply:
coolies, it wasn't here when i left my comment 😉